app/controllers/orderitems_controller.rb

```diff
@@ -0,0 +1,61 @@
+class OrderitemsController < ApplicationController
+  def add_to_cart
+    if @current_order.orderitems.where("product_id = ?", params[:id]).first
```

Comment:
I think that this `where` clause would be well-suited to be a method that would return a value in the model rather than here in the controller. You could then assign this result to a variable rather than doing a duplicate query on the next line. We want to try to reduce the number of times that we need to query the database.